runtime/runtime.go

```diff
@@ -397,6 +398,13 @@ func (rt *Runtime) Serve(ctx context.Context) error {
 		rt.logger.Error("Token authentication enabled without authorization. Authentication will be ineffective. See https://www.openpolicyagent.org/docs/latest/security/#authentication-and-authorization for more information.")
 	}
 
+	usr, err := user.Current()
```

Comment:
> I hadn't thought of that. What do Windows machines return for these calls?

I have no idea. But if it happens to be "0", it could be confusing...

Reply:
It's a bit of a different approach, but perhaps we could investigate dropping privs instead? https://sites.google.com/site/fullycapable/getting-started-with-go/using-go-to-set-uid-and-gids

Reply:
> I have no idea. But if it happens to be "0", it could be confusing...

Totally. I just did some quick research and the godoc on that field says:
> // On Windows, this is a security identifier (SID) in a string format.

The Microsoft [docs](https://docs.microsoft.com/en-us/windows/security/identity-protection/access-control/security-identifiers) on SID's say they're formatted like `S-R-X-Y1-Y2-Yn-1-Yn` ... so I don't think they can ever be just "0".

Dropping privs is an interesting approach, but that could break deployments on upgrade... so if we do that we might as well go rootless right away, and the idea here was to give people a bit of a grace period.

Reply:
Yeah let's do it then. But we should reword the logged message to say something like "running OPA as root is often unnecessary", and add a note "the official docker image will switch to rootless eventually" if we know that we're running the official docker image. A simple `RUN touch /official` in the Dockerfile and a check here would be sufficient, I think.

Reply:
Sure, I'll try that!

Reply:
```shell
 > [3/3] RUN touch /official:
#7 0.120 container_linux.go:380: starting container process caused: exec: "/bin/sh": stat /bin/sh: no such file or directory
```

![cant](https://user-images.githubusercontent.com/510711/155224664-be8ff5cb-0b9d-45a3-ab09-89d8678f7aa4.jpg)

Reply:
Haha. Let's copy in an entry file from the host then...

Reply:
... or from the builder image.

Reply:
Used an `ENV` var in the Dockerfile to check for instead.

Reply:
hah, ENV is clever 👍 

For completeness' sake:
> Dropping privs is an interesting approach, but that could break deployments on upgrade... 

The idea is that we wouldn't: you first do whatever you need to do that could require privileges (binding ports, reading files), and you drop them.

But it's tricky to get right, and the planned path forward is simpler and leads to the same destination, so ✔️

Reply:
Gotcha! Yeah, the only semi-legit reason I can think of to want to run as root would be to bind to port 443, and it seems like there are a few options to allow this for non-root users.. could be something to look into later if those who want that feel the warning is annoying.